packages/client/hmi-client/src/services/concept.ts

```diff
@@ -20,14 +21,29 @@ interface EntityMap {
 
 const curieNameCache = new Map<string, string>();
 
+const currieEntityCache = new FIFOCache<Promise<{ data: any; status: number }>>(400);
 /**
  * Get DKG entities, either single ones or multiple at a time
  */
 async function getCuriesEntities(curies: Array<string>): Promise<Array<DKG> | null> {
-	const response = await API.get(`/mira/currie/${curies.toString()}`);
-	if (response?.status === 200 && response?.data) return response.data;
-	if (response?.status === 204) console.warn('No DKG entities found for curies:', curies);
-	return null;
+	try {
+		const cacheKey = curies.toString();
+
+		let promise = currieEntityCache.get(cacheKey);
+		if (!promise) {
+			promise = API.get(`/mira/currie/${curies.toString()}`).then((res) => ({ data: res.data, status: res.status }));
+			currieEntityCache.set(cacheKey, promise);
+		}
+
+		// If a rename function is defined, loop over the first row
+		const response = await promise;
+		if (response?.status === 200 && response?.data) return response.data;
+		if (response?.status === 204) console.warn('No DKG entities found for curies:', curies);
+		return null;
+	} catch (err) {
+		logger.error(err);
+		return null;
+	}
```

Comment:
I think this setup will still get a ton of errors if the call fails - which goes into the catch block with the splendour of 100 toasters?

Reply:
Yeah we may need to tweak the axio interceptor to not spam toasters

Reply:
Yes, spamming the user with 4XX and 5XX isn't great.